src/web/core/src/main/java/org/geoserver/web/data/resource/BasicResourceConfig.java

```diff
@@ -457,4 +327,133 @@ public void validate(final Form<?> form) {
             }
         }
     }
+
+    public boolean addOtherSRS(ResourceInfo resourceInfo) {
```

Comment:
I can see this is used to perform checks in other parts of the code... What if I configure a cascaded layer via REST, will this metadata info entry be populated too?

Reply:
I will verify that

Reply:
@aaime to configure via REST, this will not be populated if its not provided inside XML, What should be best behavior if the XML tags or not found? shall the server attempt to populate them on its own if not found? because this is not just OTHER_SRS metadata map entry, its also other lists e.g available, selected, preferred image formats and remote styles. Shalll I break this into another PR for simplicity and only dead with Native CRS identification in this PR?

Reply:
All parameters that need to be present for the server to work properly should be in the main information set, not in the metadata map (it was a mistake to put them there if there is no automatic fallback), as the metadata map is for "optional extras". If not set, the server should be able to do something sane on the fly, or auto-configure the values if missing. Whether this belongs in a separate PR or not, depends on the amount of rework required.

Reply:
@aaime indeed the amount of re-work is indeed enough to qualify for separate PR.

Secondly about the metadata map is used because the functionality works across to two different types of resources (WMS and WFS-NG Feature type), hence the metadata map was used. Even if WMSLayerInfo gets this as a direct value, what will be done for WFS-NG? because its FeatureTypeInfo?

Reply:
You do two separate code paths, like for getting the native CRS.